### ISSUE TITLE
Logout-command makes NodeJS-process to exit after connection troubles

### ISSUE THREAD

**Issue by PROger4ever** [andris9/imapflow#28]
# Description
If a command was interrupted by a manual timeout (as described in #27), the process got "killed" when you try to run logout-command.

# To Reproduce
Code:
```Javascript
await client.getMailboxLock('INBOX');
```
Turn off the internet while a command is running. The client will emit an event-error, but the runnining command will never finished. We can interrupt the running in user-code as described in #27, but NodeJS-process will be suddenly stopped:
```Javascript
await _client.logout(); // HERE
```

# Expected behavior
- Running command should always give control back (by returning or throwing an exception). Command-timeouts should be set too
- Logout-command should handle interrupted state of the client correctly

# Environment
 - OS: Linux, Ubuntu 20.04
 - NodeJS: v14.9.0
 - imapflow: v1.0.48

# Probable reason
Logout-command sees a currently running command (even when it was "interrupted") and something goes wrong...

# Temporary solution in user-code
Don't run logout if any error occurred. Use `client.close()` directly, clear all resources and create a new client when something goes wrong.

**Comment by andris9** [comment]
Thank you for these reports. Could you check if this is still an issue when using the latest version of ImapFlow? I fixed the other 2 problems, so the control flow should be OK and all commands should finish.

**Comment by PROger4ever** [comment]
As far as I can check it, it works. Thanks a lot :)

**Comment by PROger4ever** [comment]
I was wrong. The problem still occurs...
Code:
```Javascript
async function _getNewClient() {
  _client = new ImapFlow(_imapOptions);

  try {
    await _wrapOperation(_client, _client.connect, OPERATION_TIMEOUT); // NOTE: no internet here
  } catch (err) {
    await _deleteClient();
    throw err;
  }

  return _client;
}

async function _deleteClient() {
  if (_client === null) {
    return;
  }

  try {
    await _client.logout(); // NOTE: process exits here
  } catch (_) {
    // NOTE: We don't care about the error, we'll close the connection force anyway
  }

  _client.close();
  _client = null;
}
```

The **_wrapOperation** listed in #27.

**Comment by shunkica** [comment]
This may be related?

```
(node:1) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'next' of undefined
    at module.exports (/opt/udocit/node_modules/imapflow/lib/commands/logout.js:20:18)
    at async ImapFlow.run (/opt/udocit/node_modules/imapflow/lib/imap-flow.js:2202:22)
    at async ImapFlow.logout (/opt/udocit/node_modules/imapflow/lib/imap-flow.js:1091:9)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
`response.next()` is called in the finally block at logout.js:20, but if an exception is thrown then response will always be undefined.

**Comment by andris9** [comment]
@shunkica  Thank you for the error stack. This should be fixed now in v1.0.56
